### ISSUE TITLE
Feature Request: TPOTOutlierDetector

### ISSUE THREAD

**Issue by Y-oHr-N** [EpistasisLab/tpot#794]
Hello,

scikit-learn 0.20 provides more consistent outlier detection API.
https://speakerdeck.com/albertcthomas/anomaly-detection-in-scikit-learn-ongoing-work-and-future-developments

- covariance.EllipticEnvelope
- svm.OneClassSVM
- ensemble.IsolationForest
- neighbors.LocalOutlierFactor

So I want an estimator that fits all outlier detection models like TPOTClassifier.

Thank you.

**Comment by joanesplazaola** [comment]
Hi, I have been checking the possibility of adding this to TPOT, but I would like to double check the procedure.

As in Novelty detection, the fit function should get both, the fit data (normal data) and a Xy validation set (mixture of normal and novel data with their labels), in order to score the validity of the evaluated models. 
Also, those labels of validation should be determined, as sklearn uses 1/-1, and many people uses 1/0.
The training time of those models can also be a problem, mostly for the One Class SVM, as the time needed may exceed easily the expected limits.
